apps/emqx_connector/src/schema/emqx_connector_schema.erl

```diff
@@ -56,7 +56,7 @@ enterprise_fields_connectors() -> [].
 
 -endif.
 
-connector_type_to_bridge_types(kafka_producer) -> [kafka_producer];
+connector_type_to_bridge_types(kafka_producer) -> [kafka, kafka_producer];
```

Comment:
Q: since this function is only used for converting v1 bridges to v2 bridges apparently, maybe only `kafka` should be in this list?

Reply:
I still need to understand better how it's used, though.  It sounds strange that both are needed in this list.

Reply:
@thalesmg , please take a look at this comment: https://github.com/emqx/emqx/pull/11842#discussion_r1376595952

Reply:
Yes, I saw it.  Still, strange that both need to exist.  Again, I don't understand the compatibility layer very well yet.  :see_no_evil:

Reply:
Me neither.. 🙈

Reply:
Yes, this is strange. We should look into why kafka_producer is needed here (seems strange). Will add an item in the TODO canvas and merge this.

Reply:
@SergeTupchiy I think the problem is that you removed `kafka` which should be there since it is the old type we upgrade from. What should not be needed is `kafka_producer`. I will merge this since it should not cause any problems to have both `kafka_producer` and `kafka` in the list and we can fix it in another PR.

Reply:
I didn't remove kafka from here, first I added kafka, so it became `[kafka, kafka_producer]`, then I changed to `[kafka]` and it broke emqx_bridge:create.